src/pages/__test__/UserProfile.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import UserProfile from '../UserProfile';
+
+describe('UserProfile Component', () => {
+  describe('Rendering', () => {
+    test('renders user profile when data is available', () => {
+      render(<UserProfile />);
+      expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
+      expect(screen.getByText('Full name')).toBeInTheDocument();
+      expect(screen.getByText('Email')).toBeInTheDocument();
+      expect(screen.getByText('Date of Birth')).toBeInTheDocument();
+      expect(screen.getByText('Registration Date')).toBeInTheDocument();
+      expect(screen.getByText('Last Login Timestamp', { selector: 'span' })).toBeInTheDocument();
+      expect(screen.getByAltText('profilePhoto')).toBeInTheDocument();
+      expect(screen.getByRole('button', { name: 'Edit Profile' })).toBeInTheDocument();
+      expect(screen.getByRole('button', { name: 'Logout' })).toBeInTheDocument();
+
+      const labelElement = screen.getByText('Full name').closest('label');
+      const inputElement = labelElement ? labelElement.querySelector('.editing-input') : null;
+
+      if (inputElement) {
+        expect(inputElement).toBeInTheDocument();
+        expect(inputElement).toHaveAttribute('type', 'text');
+      } else {
+        console.error("Input element not found");
+      }
+    });
+  });
+
+  describe('Profile Editing', () => {
+    test('enables editing when "Edit Profile" button is clicked', () => {
+      render(<UserProfile />);
+      fireEvent.click(screen.getByText('Edit Profile'));
+
+      const labelElement = screen.getByText('Full name').closest('label');
+      const inputElement = labelElement ? labelElement.querySelector('.editing-input') : null;
+
+      if (inputElement) {
+        expect(inputElement).toBeInTheDocument();
+        expect(inputElement).toHaveAttribute('type', 'text');
+        expect(inputElement).toBeEnabled();
+      } else {
+        console.error("Input element not found");
+      }
+    });
+
+    test('updates profile on "Save Changes" button click', async () => {
+      render(<UserProfile />);
+      fireEvent.click(screen.getByText('Edit Profile'));
+  
+      fireEvent.change(screen.getByLabelText('Full name'), { target: { value: 'New Name' } });
```

Comment:
In the "Profile Editing" section, you use screen.getByLabelText with different label texts. To make the tests more consistent, you might consider storing label texts in variables or constants and reusing them.

Reply:
Of course, I've already fixed it